### PR TITLE
create info log

### PR DESCRIPTION
**Pull request by siddontang**
We can use create_info_log to let RocksDB generate info LOG to other places with special name.

@BusyJay @overvenus @zhangjinpeng1987

### CODE REVIEW

File: librocksdb_sys/crocksdb/c.cc [hunk 1]
```diff
@@ -2788,4 +2788,18 @@ void crocksdb_delete_file_in_range_cf(
...
+    crocksdb_logger_t* logger = new crocksdb_logger_t;
+    if(SaveError(
+      errptr,
+      CreateLoggerFromOptions(
+        std::string(path), opts->rep, &logger->rep))) {
+      delete logger;
+      return NULL;
+    }
+
+    return logger;
```
Comment (BusyJay) [inline librocksdb_sys/crocksdb/c.cc]: How does the logger get release?

**Comment by siddontang** [comment]
PTAL @BusyJay

File: tests/test_rocksdb_options.rs [hunk 2]
```diff
@@ -132,3 +132,16 @@ fn test_set_wal_opt() {
     drop(db);
 }
 
+#[test]
+fn test_create_info_log() {
+    let path = TempDir::new("_rust_rocksdb_test_create_info_log_opt").expect("");
+    let mut opts = Options::new();
```
Comment (zhangjinpeng1987) [inline tests/test_rocksdb_options.rs]: Do we need set `log_file_time_to_roll` `max_log_file_size` and `db_log_dir` for option?

Comment (siddontang) [inline tests/test_rocksdb_options.rs]: no need here.

Comment (zhangjinpeng1987) [inline tests/test_rocksdb_options.rs]: `log_file_time_to_roll` and `max_log_file_size` control the rotate type, I think we need use them.

Comment (siddontang) [inline tests/test_rocksdb_options.rs]: yes, but here we don't need it in our test. The main purpose here is to create the LOG in another place with "LOG" name. If you want to test other log options, we should use other test.

**Comment by zhangjinpeng1987** [comment]
LGTM

**Comment by siddontang** [comment]
PTAL @BusyJay

File: src/rocksdb_options.rs [hunk 3]
```diff
@@ -741,6 +741,26 @@ impl Options {
...
+            Err(_) => {
+                return Err("Failed to convert path to CString when creating rocksdb info log"
+                    .to_owned())
+            }
+        };
+
+        unsafe {
+            let logger = ffi_try!(crocksdb_create_log_from_options(cpath.as_ptr(), self.inner));
+            crocksdb_ffi::crocksdb_options_set_info_log(self.inner, logger);
+            crocksdb_ffi::crocksdb_log_destroy(logger);
```
Comment (BusyJay) [inline src/rocksdb_options.rs]: Is it safe to destroy the logger after set to rocksdb options?

Comment (siddontang) [inline src/rocksdb_options.rs]: yes, inner logger uses share_ptr, it is fine here.

Comment (siddontang) [inline src/rocksdb_options.rs]: but we can remove this line, with just a little memory leak.

**Comment by BusyJay** [comment]
LGTM

**Pull request merged**